### ISSUE TITLE
Performance problems in Explorer + <b_table> with 2000 rows

### ISSUE THREAD

**Issue by iruizeguilaz** [bootstrap-vue/bootstrap-vue#2153]
Hi

I need help, I have a big website created with vue "vue": "^2.5.13",  "bootstrap-vue": "^2.0.0-rc.11", "bootstrap": "^3.3.7",

I was about to putiing it in production when I realised that when we have lods of rows in a table the performance in explorer (only in exporer, not in chrome, firefox, opera) is very slow.
 If I apply a big filter the performance is improved but when I remove the filter againt I have to wait around 8-10 seconds to see the table rendered, and the same happens when I try to sort a column.... it is very annoying

I have tested it with another table with 400 rows and the performance in explorer is correct, but with 1000 rows it start to wait around 4-5 in each sorting for instance...

**Is there any way to improved this without changing all my logic (conitnue having all the json in the client side)**

I deploy the application with "webpack": "^2.7.0" and production mode --> "build": "cross-env NODE_ENV=production webpack --progress --hide-modules"

this is the definition of the table

```
  <b-table striped hover responsive
                    :current-page="currentPage"
                    :items="calculatedValues"
                    :per-page="perPage"
                    :fields="tableColumns"
                    :sort-desc.sync="sortDesc"
                    :sort-by.sync="sortBy">

```

I use slots for every column like this:

  ```
<template slot="HEAD_idLicense" slot-scope="data">
                      <!-- A custom formatted header cell for field 'name' -->
                      <em>{{ $t('licenses.campo.id').toUpperCase()}}</em>
                    </template>

                    <template slot="idLicense" slot-scope="data">
                      <router-link :to="pathDetail(data.item.idLicense)" v-b-tooltip.hover :title="data.item.idLicense">  <a><i class="fas fa-pencil-alt fa-lg" style="color:#4176B4;"></i> </a></router-link>
                    </template>

```
The calculated values are the values after applying my filters...  but i think that it is not the problem becuase when I apply a big filter the response is fast

fir instace:
```
this.calculatedValues = this.licenses;
         if (this.filters.search != null && this.filters.search != ''){
             this.calculatedValues = this.calculatedValues.filter(this.contiene(this.filters.search));
         }

```
where contiene is:

```
contiene(wordToCompare) {
            return function(element) {
                return  (element.nameCompany && element.nameCompany.toUpperCase().includes(wordToCompare.toUpperCase())) ||
                (element.nameProduct &&  element.nameProduct.toUpperCase().includes(wordToCompare.toUpperCase())) ||
                (element.targetid &&  element.targetid.toUpperCase().includes(wordToCompare.toUpperCase())) ||
                  (element.comments &&   element.comments.toUpperCase().includes(wordToCompare.toUpperCase()));
            }
         }
```

**Comment by tmorehouse** [comment]
What version of IE?

IE needs several polyfills for many array processing functions (i.e. which means javascript code rather than native code)

**Comment by tmorehouse** [comment]
Also you should be using Bootstrap V4.1  for the correct CSS (not V3)

**Comment by iruizeguilaz** [comment]
**IE11**

Sorry I had in the package.json the version of boostrap 3.3.7 but in the index.html this

  <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.1.0/css/bootstrap.min.css" integrity="sha384-9gVQ4dYFwwWSjIDZnLEWnxCjeSWFphJiwGPXr1jddIhOegiu1FwO5qRGvFXOdJZ4" crossorigin="anonymous">

I have updated the version of the dependences to "bootstrap": "^4.1.3"

I will try to remove the link in the index and check why it is not using it in the main.js with the import "../node_modules/bootstrap/dist/css/bootstrap.min.css" but it the meantime I need to check the performance issue first...

**Comment by nadorator** [comment]
I have the same problem with 4000 rows even in Chrome after 4Gb of ram used it crashes. 

I tried to use ajax with an @input on the pagination but this does not work well as the sortable field are only sortered on the page rendered and no more on the whole listing of all pages.  Does any one have a solution for this ?

**Comment by tmorehouse** [comment]
are you able to try the latest dev branch? it has some fixes/improvements for b-table

**Comment by nadorator** [comment]
I'll do monday then. my bigest b-table have 30 000 rows. It is a listing of adwords keywords datas for stats and analyses purphose.  Right now I can work with it on chrome if I do not use the console and it's really slow  It would be great to update the doc to be able to use b-table with pagination and ajax with sortable and filter support.

**Comment by tmorehouse** [comment]
You can use external sorting and pagination when using provider functions, or you can disable local sorting/filtering/pagination if you want to do it externally.

**Comment by tmorehouse** [comment]
I think the latest table updates in the dev branch will improve performance. It will be available in the next release 5hat is expected to come out soon.

**Comment by nadorator** [comment]
Finally i managed to use vue worker passing parts of my data structure to
each worker (one per table) and then use ajax httprequest object  to load
all the data of the tables  from php cache files. Then i have made custom
pagination using stored data from responses and partial snapshot to render
the tables. I also made a custom sorting using also vue worker native ajax
php cache but also fusonic linq for php to sort the data quicker as i had
30 000 lines to sort. The result is really quick. I also had to manage the
php ram usage because of the json parsing using 512MB memory limit for 30
000 lines (213MB of effective cost) i hope it will help someone too.

Le mar. 11 déc. 2018 à 07:44, Troy Morehouse <notifications@github.com> a
écrit :
